core/src/main/java/com/epam/esm/dao/impl/GiftDaoImpl.java

```diff
@@ -0,0 +1,282 @@
+        List<GiftCertificateEntity> query = jdbcTemplate.query(FIND_GIFT_BY_ID_QUERY, GIFT_ROW_MAPPER, giftId);
+
+        if (query.size() != 1){
+            throw new GiftNotFoundException();
+        }
+
+        return query.get(0);
+    }
+
+    @Override
+    public List<GiftCertificateEntity> findAndSortGift(CustomSearchRequest customSearchRequest){
+        QueryArgModel queryArgModel = defineQueryAndArgs(customSearchRequest);
+        String queryString = queryArgModel.getQuery();
+        Object[] args = queryArgModel.getArgs();
+
+        List<GiftCertificateEntity> query = jdbcTemplate.query(queryString, GIFT_ROW_MAPPER, args);
+
+        for (GiftCertificateEntity giftCertificateEntity : query){
+            Set<TagEntity> tags = getTagsByGiftId(giftCertificateEntity.getId());
+            giftCertificateEntity.setTags(tags);
+        }
+        return query;
+    }
+
+    @Override
+    public GiftCertificateEntity createGift(GiftCertificateEntity entity){
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        Timestamp currentTimestamp = new Timestamp(System.currentTimeMillis());
+        jdbcTemplate.update(
+                connection -> {
+                    PreparedStatement ps =
+                            connection.prepareStatement(INSERT_GIFT_QUERY, new String[] {COLUMN_ID});
+                    ps.setString(1, entity.getName());
+                    ps.setString(2, entity.getDescription());
+                    ps.setInt(3, entity.getPrice());
+                    ps.setInt(4, entity.getDuration());
+                    ps.setTimestamp(5, currentTimestamp);
+                    ps.setTimestamp(6, currentTimestamp);
+
+                    return ps;
+                },
+                keyHolder);
+
+
+        Long giftId = (Long) keyHolder.getKey();
+
+        entity.setId(giftId);
+        entity.setCreateDate(currentTimestamp);
+        entity.setLastUpdateDate(currentTimestamp);
+
+        return entity;
+    }
+
+    @Override
+    public GiftCertificateEntity updateGift(GiftCertificateEntity giftCertificateEntity){
+        Timestamp currentTimestamp = new Timestamp(System.currentTimeMillis());
+        int rows = jdbcTemplate.update(UPDATE_BY_ID_QUERY,
+                giftCertificateEntity.getName(),
+                giftCertificateEntity.getDescription(),
+                giftCertificateEntity.getPrice(),
+                giftCertificateEntity.getDuration(),
+                giftCertificateEntity.getId());
+
+        if (rows == 0){
+            throw new GiftNotFoundException();
+        }
+
+        giftCertificateEntity.setLastUpdateDate(currentTimestamp);
+
+        return giftCertificateEntity;
+    }
+
+    @Override
+    public void deleteGiftById(Long giftId){
+        jdbcTemplate.update(DELETE_GIFT_BY_ID, giftId);
+    }
+
+    private Set<TagEntity> getTagsByGiftId(Long giftId){
+        List<TagEntity> queryTags = jdbcTemplate.query(SELECT_TAGS_BY_GIFT, TAG_ROW_MAPPER, giftId);
```

Comment:
please remove this method and get gift's tags with the help of SQL join 
remember one call to db:)